src/Storages/MergeTree/MarkRange.cpp

```diff
@@ -3,28 +3,13 @@
 namespace DB
 {
 
-namespace ErrorCodes
-{
-    extern const int LOGICAL_ERROR;
-}
-
 bool MarkRange::operator==(const MarkRange & rhs) const
 {
     return begin == rhs.begin && end == rhs.end;
 }
 
 bool MarkRange::operator<(const MarkRange & rhs) const
 {
-    /// We allow only consecutive non-intersecting ranges
-    /// Here we check whether a beginning of one range lies inside another range
-    /// (ranges are intersect)
-    const bool is_intersection = (begin <= rhs.begin && rhs.begin < end) ||
-        (rhs.begin <= begin && begin < rhs.end);
-
-    if (is_intersection)
-        throw Exception(ErrorCodes::LOGICAL_ERROR,
-        "Intersecting mark ranges are not allowed, it is a bug! First range ({}, {}), second range ({}, {})", begin, end, rhs.begin, rhs.end);
-
```

Comment:
Well, I think this check may be important for some other code.
Maybe we can write another comparator?